### PR TITLE
Ignore local pom if the jar file does not exist.

### PR DESCRIPTION
**Pull request by pstreef**

### CODE REVIEW

File: rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java [hunk 1]
```diff
@@ -516,6 +516,14 @@ public Pom download(GroupArtifactVersion gav,
                             RawPom rawPom = RawPom.parse(fis, Objects.equals(versionMaybeDatedSnapshot, gav.getVersion()) ? null : versionMaybeDatedSnapshot);
                             Pom pom = rawPom.toPom(inputPath, repo).withGav(resolvedGav);
 
+                            if("jar".equals(pom.getPackaging())){
+                                Path jar = f.toPath().getParent().resolve(gav.getArtifactId() + '-' + versionMaybeDatedSnapshot + ".jar");
+                                if (!Files.exists(jar)) {
+                                    // This is a corrupt dependency.
```
Comment (pstreef) [inline rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java:522]: It's not really corrupted. There are valid usecases for the jar to be missing (like output of mvn dependencyTree)

Comment (knutwannheden) [inline rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java:522]: As we've seen, this typically isn't the case. It just means that the jar has not been downloaded.

File: rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java [hunk 2]
```diff
@@ -516,6 +516,14 @@ public Pom download(GroupArtifactVersion gav,
                             RawPom rawPom = RawPom.parse(fis, Objects.equals(versionMaybeDatedSnapshot, gav.getVersion()) ? null : versionMaybeDatedSnapshot);
                             Pom pom = rawPom.toPom(inputPath, repo).withGav(resolvedGav);
 
+                            if("jar".equals(pom.getPackaging())){
+                                Path jar = f.toPath().getParent().resolve(gav.getArtifactId() + '-' + versionMaybeDatedSnapshot + ".jar");
```
Comment (knutwannheden) [inline rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java:520]:
```suggestion
                                Path jar = f.toPath().resolveSibling(gav.getArtifactId() + '-' + versionMaybeDatedSnapshot + ".jar");
```

File: rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java [hunk 3]
```diff
@@ -516,6 +516,14 @@ public Pom download(GroupArtifactVersion gav,
                             RawPom rawPom = RawPom.parse(fis, Objects.equals(versionMaybeDatedSnapshot, gav.getVersion()) ? null : versionMaybeDatedSnapshot);
                             Pom pom = rawPom.toPom(inputPath, repo).withGav(resolvedGav);
 
+                            if("jar".equals(pom.getPackaging())){
```
Comment (knutwannheden) [inline rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java:519]:
```suggestion
                            if (pom.getPackaging() == null || "jar".equals(pom.getPackaging())) {
```

Comment (knutwannheden) [reply rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java:519]: Note that `jar` is the default packaging. So we should also check that the jar exists, when the value is `null`.

File: rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java (hunk 1 above)
Comment (pstreef) [reply rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java:522]:
```suggestion
                                    // The jar has not been downloaded, making this dependency unusable.
```

**Review by knutwannheden — APPROVED** [review]

**Comment by joanvr** [comment]
Added check for file size == 0, since I've also found some of this broken artifacts on my local maven repository that were causing issues.
Also fixed the test that was failing and added a couple more that tests those broken artifacts

**Review by joanvr — APPROVED** [review]

**Pull request merged**